corehq/apps/hqmedia/tasks.py

```diff
@@ -214,6 +219,26 @@ def build_application_zip(include_multimedia_files, include_index_files, app,
     DownloadBase.set_progress(build_application_zip, 100, 100)
 
 
+def _check_ccz_locale_id_integrity(file_cache):
+    for path in ('default/app_strings.txt', 'suite.xml'):
+        if path not in file_cache:
+            return [_("Could not find {path} in CCZ").format(path)]
+
+    app_strings_ids = {
+        line.decode("utf-8").split('=')[0]
+        for line in file_cache['default/app_strings.txt'].splitlines()
+    }
+
+    from corehq.apps.app_manager.xform import parse_xml
+    parsed = parse_xml(file_cache['suite.xml'])
+    suite_ids = {locale.get("id") for locale in parsed.iter("locale")}
+
+    return [
+        _("Locale ID {id} present in suite.xml but not in default app strings.").format(id=id)
+        for id in (suite_ids - app_strings_ids) if id
```

Comment:
Can we add a line under method def saying something like
```
all string ids present in suite xml should be present in app_strings.
```

Why need the `if` here?

Also can the method be renamed to `find_missing_locale_ids` or something better that tells clearly about the intend of the method?

Reply:
A `None` showed up while testing; I'm guessing it's from a blank line at the end of the file.